### PR TITLE
Uae/90974/add api call and mock for claim status

### PR DESCRIPTION
**Pull request by wullaski**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- Adds `claims` to include parameter on api calls to get past appointments
- Adds `claims` to include parameter on api calls to get a past appointment
- Adds feature flag
- Adds unit tests for past appointments list and appointment details view
- Modifies confirmed appointments mocks to include claim data for testing and future development on the front-end

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
[department-of-veterans-affairs/va.gov-team#90974](https://github.com/department-of-veterans-affairs/va.gov-team/issues/90974)

## Testing done

- Unit tests that test that the `claims` string only gets added on past appointments and when the feature toggle is enabled
- Navigate to the appointments list view and see that the claims include is added to the call only on past appointments
- Navigate to a past appointment and refresh and see that the claims include is added to the api call for a single appointment
- Navigate to current appointments and see that claims is not added
- Navigate to a current appointment details view and on refresh see that claims is not added 

## What areas of the site does it impact?

Vaos/appointments app

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/vaos/appointment-list/redux/actions.js
Comment (brianseek) [inline src/applications/vaos/appointment-list/redux/actions.js:499]: are you thinking that vets-api is going to send a 404 when there is no claim found? I feel like they would sanitize the expected not found response

File: src/applications/vaos/services/mocks/v2/confirmed.json
Comment (liztownd) [inline src/applications/vaos/services/mocks/v2/confirmed.json:1497]: To clarify, if there's no claim for the appointment it will not have the claim ID and just have the "message" (although we're leaning toward including that in a `metadata` attribute, might be an object, might be a string, it's still TBD, but that way we can alert y'all to error messages and status in the appointment calls as well).

File: src/applications/vaos/services/mocks/v2/confirmed.json
Comment (liztownd) [inline src/applications/vaos/services/mocks/v2/confirmed.json:1595]: Same comment as above re: claim ID not being present if there's no claim for the associated appointment.

File: src/applications/vaos/services/mocks/v2/confirmed.json
Comment (liztownd) [inline src/applications/vaos/services/mocks/v2/confirmed.json:1360]: This is subject to change - we're not sure yet if we'll be appending the claim status as well, depending on final designs. May I ask which designs you're basing this on (so we're all on the same page, I've been waiting for finalized designs before finalizing the API association logic).

File: src/applications/vaos/services/mocks/v2/confirmed.json
Comment (wullaski) [reply src/applications/vaos/services/mocks/v2/confirmed.json:1497]: Good point about the claim id. I'll remove that. I'll leave the message as is for now until we have the API finalized. These are just so I can get the front-end logic in place for the UI.

File: src/applications/vaos/services/mocks/v2/confirmed.json
Comment (wullaski) [reply src/applications/vaos/services/mocks/v2/confirmed.json:1360]: Okay that's fine if I see an ID I think that will be enough for the proposed UI here: https://www.figma.com/design/RzugGEmu4drhCSHTyQ6hjl/Simple-mileage-only-travel-pay-claim-submission?node-id=2135-3&node-type=canvas&t=jbwHEvpeN63cY1Dw-0

**Review by liztownd — COMMENTED** [review]
I have questions about the past appointment flag (but that's really more up to the appointments team internally, mainly the comment is me voicing my confusion 🤣) and left a comment re: the different flag for the submission step. (The view details and submit SMOC are separate flags. You might want to go ahead and add the separate SMOC flag, but that's up to you.) 

Other than that the related travel pay object looks good (of course, pending changes related to the external integration, I'll let you know if things change on that front).

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx
Comment (wullaski) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx:61]: This is for when the user hits refresh on the browser and we need to re-fetch the appointment.  `isPast` needs the appointment to determine if it is a past appointment. So the only way I could see to determine if we need to fetch claim data is to check the path from the the router.

It looks like avs is added to the call on any details view when hitting refresh. Not just past appointments. So there might need to be a change made there to prevent that. @JunTaoLuo 

It works for the appointments list because there's a router that routes to `PastAppointmentsList` which calls a specific method for fetching past appointments that includes the avs argument.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx
Comment (wullaski) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx:51]: I wonder if there's two toggles because they could show claim status for existing claims but not have the ability to submit claims. 🤔 That would make sense. I'll ask in the travel pay channel.

Comment (liztownd) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx:51]: > there's two toggles because they could show claim status for existing claims but not have the ability to submit claims

Yes, that's exactly the case. I'm not sure which you're working on right now (the claim status or the submit claims) which is why I'm not sure which flag you'll need right now, but yes, we might enable the "view claim details" flag before the "submit for mileage" flag, so ideally both would be in place when it's time for the features to be flipped.

File: src/applications/vaos/services/appointment/index.js
Comment (JunTaoLuo) [inline src/applications/vaos/services/appointment/index.js:187]: Like AVS I'm okay with claims being defaulted to true for fetching single booked appointment.

File: src/applications/vaos/appointment-list/redux/actions.js
Comment (JunTaoLuo) [inline src/applications/vaos/appointment-list/redux/actions.js:489]: Since this is a single appointment, I'm also happy to have this fetch claims details by default.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx
Comment (wullaski) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx:51]: This is just for modifying the fetch for claim status the next ticket will be the UI logic and require both flags. Thanks for the clarification!

**Review by JunTaoLuo — COMMENTED** [review]
I think we can afford to be a little more aggressive with the FE calling fetch claim status (i.e. single fetches and fetching past appointments) and we'll be more selective on the BE (i.e. checking for feature toggle and checking for the appointment being a past appointment). This will bring us in line with the AVS implementation. Thoughts @simiadebowale @cferris32? 

In terms of this PR, I'm actually happy to review/approve it since it seems like it's mostly only API calls and feature toggles with no UI impact. I think once we start consuming the data and displaying it is where I'll prefer seeing a more fleshed out BE implementation/PR.

**Review by JunTaoLuo — APPROVED** [review]
Changes LGTM! But I want to wait for a go-ahead from @simiadebowale before merging just in case.

**Comment by JunTaoLuo** [comment]
We're good to merge here. I got Simi's nod of approval.

**Pull request merged**
